storage/Filesystems/FilesystemImpl.h

```diff
@@ -252,6 +253,32 @@ namespace storage
 
     }
 
+
+    class EnsureMounted : boost::noncopyable
+    {
+
+    public:
+
+	/**
+	 * Ensures that the filesystem is mounted somewhere.
+	 *
+	 * The mode is not enforced.
```

Comment:
And even if we do a second mount you cannot mount a filesystem once read-only and once read-write.